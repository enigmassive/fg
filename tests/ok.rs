use std::fs;

#[test]
fn test() {
    fs::read_dir("./tests/ok/").unwrap().for_each(|entry| {
        let entry = entry.unwrap();
        let name = entry.file_name().into_string().unwrap();
        let path = entry.path();

        let src = String::from_utf8(fs::read(path).unwrap()).unwrap();
        let mut dest = vec![];
        fg::compile(&src, &mut dest);
        let code = String::from_utf8(dest).unwrap();
        insta::assert_snapshot!(name, code);
    })
}
