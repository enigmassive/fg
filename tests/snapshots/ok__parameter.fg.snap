---
source: tests/ok.rs
expression: code
---
package parameter

func noParameters() {
}

func withParameter(a int) {
}

func withParameters(a int, b bool) {
}

func withUnnamedParameter(int) {
}

func withUnnamedParameters(int, bool) {
}

func withReturn(a int, b string) (c bool) {
}

func withReturns(a int, b string) (c bool, d error) {
}

func withUnnamedReturn(a int, b string) (bool) {
}

func withUnnamedReturns(a int, b string) (bool, error) {
}

func withNakedReturn(a int, b string) (bool) {
}
